Create slash command for /poll widget.

showell opened the pull request:
This is the most versatile flavor of a slash command in terms of allowing devs to build multi-user widgets like the survey/poll widget here.  (There's also a tic-tac-toe game to illustrate how the framework works for more than just surveys.)

The key pieces here are:

- **submessages** = data transport (similar conceptually to reactions, but more generic)
- **widget.py** = slash command dispatch
- **widgetize.js** = front end for interactive widgets

Now I say this code is versatile, but what I really mean is that it has lots and lots of possible applications.  What it doesn't specifically handle is ephemeral messages--that's a somewhat separate project, although it will share widget.py, so it would be nice to get that merged.

timabbott commented:
A few requests:
* Can you add a `settings.DEVELOPMENT` guard for (1) the submessage endpoints and (2) at the top of `enable_widgets_for_message`?  I imagine we'll remove that on `tabbott/master` (for chat.zulip.org) to make testing easier, but like other experimental features, I want it off for production environments to start so I don't need to think through the implications too deeply :).  
* I'm not sure `sender` is the right name for the user field on this table; why did you pick that over `user_id`?  We can discuss in person since we're both here at the PyCon sprints.
* It looks like the query counts are off somewhere for the tests in e.g. `test_subs.py` (probably just a rebase conflict).

showell commented:
I re-pushed so that @lonerz can rebase off of this, but I haven't addressed all feedback yet.

I'm starting to think sender -> user is not an improvement.  "sender" is specific and correct, just slightly confusing out of context, whereas "user" is vague and still likely to lead to confusion.

timabbott commented:
Merged, thanks @showell!
